### ISSUE TITLE
Reading TOAs from a pickle file fails on Macs

### ISSUE THREAD

**Issue by paulray** [nanograv/PINT#29]
For some reason, the pickle files don't read on Macs.  The error looks like this:
```
INFO: Reading toas from 'examples/NGC6440E.tim.pickle.gz'...
 [pint.toa]
Traceback (most recent call last):
  File "examples/fit_NGC6440E.py", line 18, in <module>
    t = pint.toa.get_TOAs(timfile)
  File "/Users/paulr/src/PINT/pint/toa.py", line 33, in get_TOAs
    t = TOAs(timfile,usepickle=usepickle)
  File "/Users/paulr/src/PINT/pint/toa.py", line 223, in __init__
    self.read_toa_file(toafile, usepickle=usepickle)
  File "/Users/paulr/src/PINT/pint/toa.py", line 518, in read_toa_file
    tmp = cPickle.load(gzip.open(filename+ext, 'rb'))
ValueError: non-string names in Numpy dtype unpickling
```

This should be fixed, and a unit test designed specifically to test the TOA pickle/unpickle process should be added.  All other unit tests should _not_ use pickling.

**Comment by paulray** [comment]
This may be related to this issue in astropy: https://github.com/astropy/astropy/issues/4098

**Comment by paulray** [comment]
Jing reports that this issue does not occur when using numpy 1.8.

**Comment by luojing1211** [comment]
I run my test. It returns a different issue. 
>>>
INFO: Reading toas from 'examples/NGC6440E.tim.pickle.gz'...
 [pint.toa]
Traceback (most recent call last):
  File "examples/fit_NGC6440E.py", line 21, in <module>
    rs = pint.residuals.resids(t, m).phase_resids
  File "/Users/jingluo/Research_codes/PINT/pint/residuals.py", line 12, in __init__
    self.chi2 = self.calc_chi2()
  File "/Users/jingluo/Research_codes/PINT/pint/residuals.py", line 52, in calc_chi2
    return ((self.time_resids / self.toas.get_errors()).decompose()**2.0).sum()
  File "/Users/jingluo/anaconda/lib/python2.7/site-packages/astropy-1.1.dev13975-py2.7-macosx-10.5-x86_64.egg/astropy/units/quantity.py", line 842, in __div__
    return self.__truediv__(other)
  File "/Users/jingluo/anaconda/lib/python2.7/site-packages/astropy-1.1.dev13975-py2.7-macosx-10.5-x86_64.egg/astropy/units/quantity.py", line 821, in __truediv__
    return super(Quantity, self).__truediv__(other)
ValueError: operands could not be broadcast together with shapes (62,) (0,)
>>>
There is no data in the pickle file.

**Comment by vhaasteren** [comment]
Can we confirm that this is solved? For me, with the development version of Astropy, and numpy 1.10.1 it is no longer an issue.

**Comment by scottransom** [comment]
This should be fixed on all platforms with the latest astropy.  I think that the fix is in the new release candidate for v1.1

**Comment by paulray** [comment]
I finally got a chance to test it on my Mac with the dev version of astropy. Test runs fine now.  Closing...
